go-controller/pkg/ovn/gress_policy.go

```diff
@@ -0,0 +1,193 @@
+package ovn
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	v1 "k8s.io/api/core/v1"
+	knet "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type gressPolicy struct {
+	namespace  string
+	name       string
+	policyType knet.PolicyType
+	idx        int
+
+	// peerAddressSet points to the addressSet that holds all peer pod
+	// IP addresess.
+	peerAddressSet *addressSet
+
+	// nsAddressSets holds the names of all namespace address sets
+	nsAddressSets sets.String
+
+	// sortedPeerAddressSets has the sorted peerAddressSets
+	sortedPeerAddressSets []string
+
```

Comment:
add a port group association here?

Reply:
@trozet I'm not quite sure what you mean. The port group is owned by the NamespacePolicy object in policy.go. But there's already kinda an association higher up with the portGroupUUID member.

Reply:
I don't think that is cached right? I was mainly thinking about the association of a policy should really be a port group (not an address set).
so the association in OVN is like this, and I'm thinking of having parity in our code for how we represent and associate these objects (ignore the type holding multiple values, could be map, set, whatever; point is just to show the underlying type association):
```
type portGroup {
    ACLs []acl{}
    Ports []string{}
}

type acl {
  action string
  direction string
  match string //contains references to address set
  priority string
  addressSets []addressSet{} // links associated address sets, without embedding in hard to search string
}

```
Address sets could also have reverse linkage to the the ACLs they are currently referencing as well as the port group those ACLs are in, but that's a lot of relationship maintenance.

So that covers OVN entities and their relationships, but then we have to link that to the k8s concept of a gress policy so:

```
type gressPolicy struct {
     namespace string
     ...
     portGroups []portGroup{} //not sure if this really is a list or not, but you get the idea
```
So if we are going to cache, cache port groups and ACLs too. Then when there is an update/delete, we simply delete  the uuid from ovn, and there is no need to search or translate anything. Down the road if we want to do the idea of storing this kind of "ovn-k8s" data into CRDs we can re-use the same code and set it in CRD types instead of a local cache.

Reply:
Sure? I don't think that affects this PR though...